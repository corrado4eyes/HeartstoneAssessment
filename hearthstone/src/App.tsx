import React from 'react';
import './App.css';
import ServiceFactory from './services/serviceFactory';
import Card from './model/card';
import inversifyContainer from '../config/inversify';
import {Container, Row, Col, Image} from 'react-bootstrap';

const serviceFactory = new ServiceFactory(inversifyContainer);

interface State {
    cards: any[]
}

interface Props{}
class App extends React.PureComponent<Props, State> {
constructor(props: Props){
      super(props)
      this.state = {
        cards: []
      }
    }

    getData = (): any => {
        return serviceFactory.getCardService().getAll()
        .then((cards: Card[]) => this.setState({cards: cards}))
        .catch((error: any) => {return error});
    }

    componentDidMount(){
      this.getData()
    }

    render(){
        return (
            <Container>
                <Row>
                    {
                        this.state.cards.map((el: Card) => {
                            return (el.img || el.imgGold) ? (
                                <Col key={el.cardId} className='card' md="4">
                                    <Image src={el.img}/>
                                </Col>
                            ) : undefined
                        })
                    }
                </Row>
            </Container>
        );
    }
}

export default App;
