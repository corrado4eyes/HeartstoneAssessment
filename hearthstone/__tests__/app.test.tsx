import React from 'react';
import App from '../src/App';
import {shallow, ShallowWrapper} from 'enzyme'
import { dummyCardArray } from './__mocks__/mockObjects';
import Card from '../src/model/card';


describe('App', () => {
    const cards: Card[] = dummyCardArray
    it('renders n elements in an array', () => {
        const component: ShallowWrapper = shallow(<App/>);
        component.setState({cards: cards})
        const divs: ShallowWrapper = component.find('.card');
        expect(divs).toHaveLength(cards.length);
    });

    it('Snapshot testing', () => {
            const component: ShallowWrapper = shallow(<App/>);
            component.setState({cards: cards})
            expect(component).toMatchSnapshot();
    });
});
