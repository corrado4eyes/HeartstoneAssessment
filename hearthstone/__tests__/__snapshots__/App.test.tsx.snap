// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot testing 1`] = `
<Fragment>
  <div
    className="card"
    key="what a dummy card!"
  >
    corrado4eyes
  </div>
</Fragment>
`;
